tests/jest/helpers/main.js

```diff
@@ -0,0 +1,221 @@
+const ChatkitServer = require("@pusher/chatkit-server").default
+const uuid = require("uuid/v4")
+
+const config = require("./config/production")
+
+async function defaultBeforeAll(roleName) {
+  page.on("console", async msg => {
+    const argsWithRichErrors = await Promise.all(
+      msg
+        .args()
+        .map(arg =>
+          arg
+            .executionContext()
+            .evaluate(arg => (arg instanceof Error ? arg.message : arg), arg),
+        ),
+    )
+    console.log(...argsWithRichErrors)
+  })
+
+  await page.addScriptTag({ path: "./dist/web/chatkit.js" })
+  await page.evaluate(config => {
+    window.ChatManager = Chatkit.ChatManager
+    window.TokenProvider = Chatkit.TokenProvider
+    window.config = config
+
+    window.makeChatManager = user =>
+      new ChatManager({
+        instanceLocator: config.INSTANCE_LOCATOR,
+        userId: user.id,
+        logger: {
+          verbose: () => {},
+          debug: () => {},
+          info: console.info,
+          warn: console.warn,
+          error: console.error,
+        },
+        tokenProvider: new TokenProvider({
+          url: config.TOKEN_PROVIDER_URL,
+        }),
+      })
+  }, config)
+
+  await makeGlobalRole(roleName)
+}
+
+async function defaultAfterAll(roleName) {
+  removeGlobalRole(roleName)
+}
+
+function makeUser(roleName) {
+  const server = new ChatkitServer({
+    instanceLocator: config.INSTANCE_LOCATOR,
+    key: config.INSTANCE_KEY,
+  })
+  return server
+    .createUser({
+      id: uuid(),
+      name: uuid(),
+    })
+    .then(res => {
+      return server
+        .assignGlobalRoleToUser({
+          userId: res.id,
+          name: roleName,
+        })
+        .then(() => res)
+    })
+    .then(res => ({
+      id: res.id,
+      name: res.name,
+    }))
+}
+
+function makeRoom({ members }) {
+  return new ChatkitServer({
+    instanceLocator: config.INSTANCE_LOCATOR,
+    key: config.INSTANCE_KEY,
+  })
+    .createRoom({
+      id: uuid(),
+      name: uuid(),
+      creatorId: members[0].id,
+      userIds: members.map(m => m.id),
+    })
+    .then(res => ({
+      id: res.id,
+      name: res.name,
+    }))
+}
+
+function makeGlobalRole(name) {
+  return new ChatkitServer({
+    instanceLocator: config.INSTANCE_LOCATOR,
+    key: config.INSTANCE_KEY,
+  })
+    .createGlobalRole({
+      name: name,
+      permissions: [
+        "message:create",
+        "room:join",
+        "room:leave",
+        "room:members:add",
+        "room:members:remove",
+        "room:get",
+        "room:create",
+        "room:update",
+        "room:delete",
+        "room:messages:get",
+        "room:typing_indicator:create",
+        "presence:subscribe",
+        "user:get",
+        "user:rooms:get",
+        "file:get",
+        "file:create",
+        "cursors:read:get",
+        "cursors:read:set",
+      ],
+    })
+    .then(() => {
+      return
+    })
+    .catch(err => {
+      return
```

Comment:
this throws away the error, is this okay? If you don't want to throw it away you can just remove the catch block

Reply:
Probably not.. It was for cases where the teardown didn't work, but perhaps we can just delete it before.

Reply:
I decided to keep this for now, as otherwise we'll need to handle the case where the role is already defined (which should probably be handled differently)

Added a comment